memguard.go

```diff
@@ -38,13 +39,21 @@ Similarly, if a function is given a LockedBuffer that has been
 destroyed, the call will return an ErrDestroyed.
 */
 type LockedBuffer struct {
+	*lockedBuffer
```

Comment:
Could you rename this to something more descriptive, maybe `container`? Just to reduce possible confusion when skimming the codebase.